### DISCUSSION TITLE
Next.js i18n and Markdoc

### DISCUSSION THREAD

**Discussion by KristianOellegaard** [markdoc/markdoc#330] [Q&A] [answered]
Hey,

We have enabled the i18n feature of Next.js but are struggling to pass on the current language and country to Markdoc. Any idea how to expose variables to Markdoc with the Next.js implementation that depends on the request (so that you can read the language and country).

Kind regards,
Kristian

**Comment by mfix-stripe** [comment, marked as answer]
@KristianOellegaard you can do so in your `_app.js` file (here: https://github.com/markdoc/docs/blob/main/pages/_app.js#L157)

You can also just set the value in a React Context and read it within your React components directly.

**Reply by KristianOellegaard** [reply]
Thank you for the quick reply!

I can't figure out how to do it. I tried:

```jsx
<Component country={"US"} {...pageProps} />
```

and

```javascript
export default function App({ Component, pageProps }) {
  let title = pageProps.markdoc?.frontmatter.title
  let router = useRouter()
  let isDocsPage = router.pathname.startsWith("/docs")

  let pageTitle =
    pageProps.markdoc?.frontmatter.pageTitle || (
        isDocsPage && `${pageProps.markdoc?.frontmatter.title} - Site Docs`
    ) || `${pageProps.markdoc?.frontmatter.title} - Site`

  pageProps.markdoc.country = "US";

  return (
    <>
      <Head>
        <title>{pageTitle}</title>
      </Head>
      <Layout>
        <Component {...pageProps} />
      </Layout>
    </>
  )
}

```

without any result

I'm trying to achieve the following in my Markdown
```markdown
{% if equals($country, "US") %}
```

**Reply by mfix-stripe** [reply]
Ahh I see. For variables you want to access within the Markdoc document, you can export a `variables` object from `markdoc/config.js`, like:
```js
// markdoc/config.js

export const variables = {
  country: 'US'
}
```

If you need to update that value at runtime, I would encourage you to move that state into _React_, not the Markdoc document.

However, with Next.js, you can split the difference between these solutions by creating a JS page file, exporting `getStaticProps` or `getServerSideProps` like this:

```js
// Import your .md file
import Component, {getStaticProps} from './_index.md';

export async function getServerSideProps(ctx) {
  return getStaticProps({
    ...ctx,
    variables: {
      country: 'US',
    },
  });
}

export default Component;
```
